core/src/main/java/org/elasticsearch/indices/IndicesQueryCache.java

```diff
@@ -45,13 +45,17 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Predicate;
 
 public class IndicesQueryCache extends AbstractComponent implements QueryCache, Closeable {
 
     public static final Setting<ByteSizeValue> INDICES_CACHE_QUERY_SIZE_SETTING = Setting.byteSizeSetting(
             "indices.queries.cache.size", "10%", Property.NodeScope);
     public static final Setting<Integer> INDICES_CACHE_QUERY_COUNT_SETTING = Setting.intSetting(
             "indices.queries.cache.count", 10000, 1, Property.NodeScope);
+    // enables caching on all segments instead of only the larger ones, for testing only
+    public static final Setting<Boolean> INDICES_QUERIES_CACHE_ALL_SEGMENTS_SETTING = Setting.boolSetting(
```

Comment:
Currently I am using this setting in EsIntegTestCase so that caching actually happens in our integration tests (otherwise we would never cache since we only cache on segments that have more than 10k docs), but it feels wrong to install the InternalSettingsPlugin in the base class. So I believe I could either give up on enabling caching in tests, leave this setting registered by default, or create a dedicated plugin for this setting?